.github/workflows/release.yml

```diff
@@ -98,3 +98,80 @@ jobs:
         run: |
           make docker.buildx.migrate
           make buildx.push.migrate
+
+  unit-tests:
+    name: unit tests 
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Set up Go 1.14
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.14.7
+      id: go
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: verify license
+      run: make check-license 
+
+    - name: verify dependencies
+      run: make deps
+
+    - name: verify tests
+      run: make test
+
+  migration-e2e:
```

Comment:
There is a change that this tests can run before the actual image is pushed, since it is dependent only on the unit tests. This will be an issue only for release, as for other cases, the ci tag will always be present.

Reply:
For now I am building the local images in the sanity script itself with the appropriate tags. I will make those changes in upcoming PRs.

Reply:
Also added proper `needs: []` in the build and release workflow